### PR TITLE
Refactor ThreadManager to split out ThreadChannel logic

### PR DESCRIPTION
**Pull request by egnor**
Still needs more testing (both a unit test and manual testing), but please take a look to verify this improves the complexity that was hard to understand previously. No functionality changes.

### CODE REVIEW

File: escape_roomba/event_logger.py [hunk 1]
```diff
@@ -6,43 +6,40 @@
 
 from escape_roomba.format_util import fid, fobj
 
+logger = logging.getLogger('bot.events')
```
Comment (anaulin) [inline escape_roomba/event_logger.py]: In the other classes you used a private `_logger`, but not here. Seems inconsistent, unless you plan other modules to use this events logger as well?

File: escape_roomba/thread_manager.py [hunk 2]
```diff
@@ -80,101 +59,80 @@ async def _on_ready(self):
...
-    #
-    # Discord channel event listeners (above)
-    #  🡲  _async_fetch_recents (catches up to recent channel history on start)
-    #      🡢  ( _async_check_message for recent messages - see above )
-    #  🡲  _async_forget_channel (removes _Thread object from tracking)
-    #  🡲  _async_check_channel (filters update, acquires lock)
-    #     ┄┄┄┄┄┄┄ methods below run with the lock held ┄┄┄┄┄┄┄┄
-    #      🡲  _async_fetch_locked_intro (loads first 2 messages)
-    #          🡢  ( _async_fetch_locked_message - see section above )
-    #
```
Comment (anaulin) [inline escape_roomba/thread_manager.py:172]: Did this documentation disappear altogether? Maybe it's my 10:30 PM brain, but I don't see it, not sure if that was intentional.

**Review by anaulin — APPROVED** [review]
I wasn't sure from your PR description if you were meaning to add tests to this PR after a first review pass, or not, but generally this seems good enough to merge. I haven't reviewed the code here in super-fine detail, but in structure it does seem like an improvement.

(Let me know if you want me to do a more fine-combed review, too. Happy to do that.)

**Comment by egnor** [comment]
> I wasn't sure from your PR description if you were meaning to add tests to this PR after a first review pass, or not, but generally this seems good enough to merge. I haven't reviewed the code here in super-fine detail, but in structure it does seem like an improvement.

I figure that would be a follow-up at some point...

**Pull request merged**